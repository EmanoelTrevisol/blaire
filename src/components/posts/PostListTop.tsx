import React from 'react';
import {
  View,
  TouchableOpacity,
  TextInput,
  StyleSheet,
  Platform,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

export interface IProps {
  searchText?: string;
  onSearchChange: (search: string) => void;
  onAddClick: () => void;
}

const PostListTop = (props: IProps) => {
  const { searchText, onSearchChange, onAddClick } = props;

  return (
    <View style={stl.top}>
      <View style={stl.searchBar}>
        <Icon style={stl.searchBarIcon} name="search" />
        <TextInput
          style={stl.searchBarInput}
          value={searchText}
          onChangeText={onSearchChange}
          placeholder="Pesquise um post"
          placeholderTextColor="#000"
        />
      </View>
      <View style={stl.newPost}>
        <TouchableOpacity onPress={onAddClick}>
          <Icon style={stl.newPostIcon} name="plus" />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const stl = StyleSheet.create({
  top: {
    ...Platform.select({
      android: {
        marginTop: 15,
      },
    }),
    marginHorizontal: 15,
    marginBottom: 15,
    height: 40,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  searchBar: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    height: 40,
    marginRight: 10,
    borderWidth: 1,
    borderColor: '#DADADA',
    borderRadius: 20,
  },
  searchBarIcon: {
    padding: 10,
    fontSize: 16,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  searchBarInput: {
    flex: 1,
    paddingVertical: 10,
  },
  newPost: {
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  newPostIcon: {
    fontSize: 20,
  },
});

export default PostListTop;
