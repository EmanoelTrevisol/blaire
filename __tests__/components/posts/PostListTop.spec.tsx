import React from 'react';
import PostListTop, { IProps } from '@components/posts/PostListTop';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import { TouchableOpacity } from 'react-native';

let props: IProps;

describe('PostListTop.tsx', () => {
  const renderComp = (cpProps: IProps) => {
    return render(<PostListTop {...cpProps} />);
  };

  beforeEach(() => {
    props = {
      searchText: '',
      onSearchChange: jest.fn(),
      onAddClick: jest.fn(),
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  test('matches snapshot', () => {
    const { toJSON } = renderComp(props);

    expect(toJSON()).toMatchSnapshot();
  });

  test('calls onSearchChange', async () => {
    const search = 'Coffee';
    const { getByPlaceholderText } = renderComp(props);

    const textInput = getByPlaceholderText('Pesquise um post');

    await waitFor(() => fireEvent.changeText(textInput, search));

    expect(props.onSearchChange).toHaveBeenCalledWith(search);
  });

  test('render with searchText prop on input', async () => {
    const searchText = 'Coffee';
    props.searchText = searchText;
    const { getByPlaceholderText } = renderComp(props);

    const textInput = getByPlaceholderText('Pesquise um post');

    expect(textInput.props.value).toBe(searchText);
  });

  test('calls onAddClick when icon is clicked', async () => {
    const { UNSAFE_getAllByType } = renderComp(props);

    const addPostButton = UNSAFE_getAllByType(TouchableOpacity);

    await waitFor(() => fireEvent.press(addPostButton[0]));

    expect(addPostButton.length).toBe(1);
    expect(props.onAddClick).toHaveBeenCalledTimes(1);
  });
});
