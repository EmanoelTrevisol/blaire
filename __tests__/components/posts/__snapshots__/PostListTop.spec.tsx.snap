// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListTop.tsx matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-between",
      "marginBottom": 15,
      "marginHorizontal": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#DADADA",
        "borderRadius": 20,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "flex-start",
        "marginRight": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "alignItems": "center",
            "fontSize": 16,
            "height": 40,
            "justifyContent": "center",
            "padding": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[MockFunction]}
      placeholder="Pesquise um post"
      placeholderTextColor="#000"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "paddingVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
